### DISCUSSION TITLE
Github's markdown rendering should support tabulating from CSV (inline or in files)

### DISCUSSION THREAD

**Discussion by masklinn** [community/community#114999] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

GFM supports tables which is nice, github can also render the content of a CSV file in a repository to a table, which is also nice.

What is missing is the intersection of the two: in order to add tabular data sourced from a CSV in an issue, a readme, a wiki page, or a discussion, it's necessary to use offline tools or possibly dodgy websites to convert the source CSV into a markdown table which GFM does understand, and repeat every time the data has to be updated.

Having something like a "table-csv" fenced code block type would be really nice.

**Comment by Bhupesh-V** [comment]
Plus one on this, even a simple CSV file link render (just like for PR reviews, and image previews) would be a nice addition.

**Reply by r-lyeh** [reply]
Ideally, users would just type following `csv` tag in a ``` section to have the CSV viewer embedded in GFM. Like:

```csv
A,B,C
1,2,3
```

**Reply by masklinn** [reply]
CSV is probably a bad choice because Linguist might add CSV support one day, or people might tag the blocks as CSV simply to indicate that they're that without wanting those to be table-formatted. Hence the suggestion of a different tag for that feature.

Technically this could also be done by providing formatting parameters in the [info string](https://spec.commonmark.org/0.12/#info-string) e.g. `csv format=table` but I don't know that github uses that anywhere at the moment so that might be a much bigger change.

**Reply by bbrewington** [reply]
In case it helps for reference, Linguist does have CSV, as `type: data`:

https://github.com/github-linguist/linguist/blob/94436d2416f841d16d5a8b38cb6777c634a2dc56/lib/linguist/languages.yml#L969-L976

**Comment by mgifford** [comment]
I'm a bit confused as to why this isn't enabled for GitHub Pages - https://jekyllrb.com/tutorials/csv-to-table/
